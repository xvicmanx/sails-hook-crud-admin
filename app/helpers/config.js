import { queryValue } from './object';

export const getConfig = () => {
  return typeof window !== 'undefined' && window.crudAdminConfig ?
  window.crudAdminConfig : {};
};

export const getLabel = (prop, defaultValue) => {
  const config = getConfig();
  return queryValue(
    config,
    `general.labels.${prop}`,
    defaultValue
  );
};

export const getModelRelatedValue = (query, defaultValue) => {
  const config = getConfig();
  return queryValue(
    config,
    `models.${query}`,
    defaultValue
  );
};

export const getButtonText = (button, defaultValue) => {
  const config = getConfig();
  return queryValue(
    config,
    `general.buttons.${button}`,
    defaultValue
  );
};

export default {
  getConfig,
  getLabel,
  getButtonText,
  getModelRelatedValue,
};
