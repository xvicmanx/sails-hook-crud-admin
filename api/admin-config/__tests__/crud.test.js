import config from '../crud';

describe('api::admin-config', () => {
  describe('CRUD configuration', () => {
    test('it is configured as expected', () => {
      expect(config).toMatchSnapshot();
    });
  });
});
