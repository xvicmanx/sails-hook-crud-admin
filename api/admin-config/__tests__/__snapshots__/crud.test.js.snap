// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api::admin-config CRUD configuration it is configured as expected 1`] = `
Object {
  "models": Object {
    "crudaction": Object {
      "fields": Object {
        "createdAt": Object {
          "icon": "calendar",
        },
        "id": Object {
          "icon": "hashtag",
        },
        "name": Object {
          "icon": "wrench",
        },
        "updatedAt": Object {
          "icon": "calendar",
        },
      },
      "icon": "wrench",
      "label": "CRUD Action",
    },
    "crudasset": Object {
      "fields": Object {
        "createdAt": Object {
          "icon": "calendar",
        },
        "fileDirectory": Object {
          "icon": "folder",
        },
        "hash": Object {
          "icon": "hashtag",
        },
        "id": Object {
          "icon": "hashtag",
        },
        "model": Object {
          "icon": "globe",
        },
        "name": Object {
          "icon": "file text",
        },
        "type": Object {
          "icon": "file",
        },
        "updatedAt": Object {
          "icon": "calendar",
        },
        "url": Object {
          "icon": "linkify",
        },
      },
      "icon": "image",
    },
    "crudgroup": Object {
      "fields": Object {
        "createdAt": Object {
          "icon": "calendar",
        },
        "id": Object {
          "icon": "hashtag",
        },
        "name": Object {
          "icon": "users",
        },
        "rights": Object {
          "icon": "key",
          "valueTemplate": "
            <ul>
              <% _.forEach(rights, function(right) { %>
                <li> <%= right.name %> </li>
              <% }); %>
            </ul>
          ",
        },
        "updatedAt": Object {
          "icon": "calendar",
        },
      },
      "icon": "users",
      "label": "CRUD Group",
      "valueTemplate": "<%= crudgroup.name %>",
    },
    "crudresource": Object {
      "fields": Object {
        "createdAt": Object {
          "icon": "calendar",
        },
        "id": Object {
          "icon": "hashtag",
        },
        "name": Object {
          "icon": "globe",
        },
        "updatedAt": Object {
          "icon": "calendar",
        },
      },
      "icon": "globe",
      "label": "CRUD Resource",
    },
    "crudright": Object {
      "fields": Object {
        "action": Object {
          "icon": "wrench",
          "valueTemplate": "<%= action.name %>",
        },
        "createdAt": Object {
          "icon": "calendar",
        },
        "id": Object {
          "icon": "hashtag",
        },
        "name": Object {
          "icon": "key",
        },
        "resource": Object {
          "icon": "globe",
          "valueTemplate": "<%= resource.name %>",
        },
        "updatedAt": Object {
          "icon": "calendar",
        },
      },
      "icon": "key",
      "label": "CRUD Right",
      "valueTemplate": "<%= crudright.name %>",
    },
    "cruduser": Object {
      "fields": Object {
        "createdAt": Object {
          "icon": "calendar",
        },
        "groups": Object {
          "icon": "users",
          "valueTemplate": "
            <ul>
              <% _.forEach(groups, function(group) { %>
                <li> <%= group.name %> </li>
              <% }); %>
            </ul>
          ",
        },
        "id": Object {
          "icon": "hashtag",
        },
        "password": Object {
          "icon": "key",
          "renderer": "password",
          "valueTemplate": "*******",
        },
        "updatedAt": Object {
          "icon": "calendar",
        },
        "username": Object {
          "icon": "user",
        },
      },
      "icon": "user",
      "label": "CRUD User",
      "valueTemplate": "<%= cruduser.username %>",
    },
  },
}
`;
